[CI] SingleNodeTests testThatLoadingWithNonExistingIndexWorks failing

**Build Scans:**
- [elasticsearch-intake #39055 / part2](https://gradle-enterprise.elastic.co/s/qieldblse4vug)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:watcher:internalClusterTest" --tests "org.elasticsearch.xpack.watcher.test.integration.SingleNodeTests.testThatLoadingWithNonExistingIndexWorks" -Dtests.seed=E4FD6C94C27D22B0 -Dtests.locale=lkt-Latn-US -Dtests.timezone=America/Grenada -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.watcher.test.integration.SingleNodeTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testThatLoadingWithNonExistingIndexWorks'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Shard [.ds-.watcher-history-18-2026.05.05-000001][0] is still locked after 5 sec waiting
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testThatLoadingWithNonExistingIndexWorks
- [main] 1 consecutive failures in step part2
- [main] 1 consecutive failures in pipeline elasticsearch-intake

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -545,9 +545,6 @@ tests:
 - class: org.elasticsearch.xpack.stateless.StatelessHollowIndexShardsIT
   method: testStressWithoutFailures
   issue: https://github.com/elastic/elasticsearch/issues/148310
-- class: org.elasticsearch.xpack.watcher.test.integration.SingleNodeTests
-  method: testThatLoadingWithNonExistingIndexWorks
-  issue: https://github.com/elastic/elasticsearch/issues/148323
 - class: org.elasticsearch.xpack.stateless.lucene.BlobCacheIndexInputTests
   method: testWithByteBufferSlicesReturnsFalseAfterEviction
   issue: https://github.com/elastic/elasticsearch/issues/148330

--- a/x-pack/plugin/watcher/src/internalClusterTest/java/org/elasticsearch/xpack/watcher/test/integration/SingleNodeTests.java
+++ b/x-pack/plugin/watcher/src/internalClusterTest/java/org/elasticsearch/xpack/watcher/test/integration/SingleNodeTests.java
@@ -12,7 +12,9 @@ import org.elasticsearch.cluster.metadata.IndexMetadata;
 import org.elasticsearch.protocol.xpack.watcher.PutWatchResponse;
 import org.elasticsearch.rest.RestStatus;
 import org.elasticsearch.test.ESIntegTestCase.ClusterScope;
+import org.elasticsearch.xpack.core.watcher.execution.TriggeredWatchStoreField;
 import org.elasticsearch.xpack.core.watcher.history.HistoryStoreField;
+import org.elasticsearch.xpack.core.watcher.transport.actions.delete.DeleteWatchRequestBuilder;
 import org.elasticsearch.xpack.core.watcher.transport.actions.put.PutWatchRequestBuilder;
 import org.elasticsearch.xpack.core.watcher.watch.Watch;
 import org.elasticsearch.xpack.watcher.test.AbstractWatcherIntegrationTestCase;
@@ -23,6 +25,7 @@ import java.util.concurrent.TimeUnit;
 
 import static org.elasticsearch.test.ESIntegTestCase.Scope.SUITE;
 import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertHitCount;
 import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertResponse;
 import static org.elasticsearch.xpack.watcher.actions.ActionBuilders.loggingAction;
 import static org.elasticsearch.xpack.watcher.client.WatchSourceBuilders.watchBuilder;
@@ -72,6 +75,16 @@ public class SingleNodeTests extends AbstractWatcherIntegrationTestCase {
                 searchResponse -> assertThat(searchResponse.getHits().getTotalHits().value(), is(greaterThanOrEqualTo(1L)))
             );
         }, 30, TimeUnit.SECONDS);
+
+        // Delete watch so that it can't refire during cleanup, recreating the watch-history index after it has been deleted.
+        assertThat(new DeleteWatchRequestBuilder(client()).setId(watchId).get().isFound(), is(true));
+
+        // and drain any tasks that may have been triggered but not yet executed, since they will still record history
+        assertBusy(() -> {
+            indicesAdmin().prepareRefresh(TriggeredWatchStoreField.INDEX_NAME).get();
+            assertHitCount(prepareSearch(TriggeredWatchStoreField.INDEX_NAME).setSize(0), 0L);
+        });
+
     }
 
 }

